src/qt3utils/datagenerators/princeton.py

```diff
@@ -0,0 +1,313 @@
+    def exposure_time(self):
+        """
+        Returns the single frame exposure time (in ms).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.ShutterTimingExposureTime)
+
+    @exposure_time.setter
+    def exposure_time(self, ms):
+        """
+        Sets the single frame exposure time to be ms (in ms).
+        """
+        return self.light.set(lf.AddIns.CameraSettings.ShutterTimingExposureTime, ms)
+
+    @property
+    def sensor_temperature(self):
+        """
+        Returns the current sensor temperature (in celcius).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureReading)
+
+    @property
+    def temperature_sensor_setpoint(self):
+        """
+        Returns the sensor setpoint temperature (celcius).
+        """
+        return self.light.get(lf.AddIns.CameraSettings.SensorTemperatureSetPoint)
+
+    @temperature_sensor_setpoint.setter
+    def temperature_sensor_setpoint(self, deg_C):
+        """
+        Sets the sensor target temperature (in degrees Celsius) to deg_C.
+        This function retrieves image data, with particular attention to the camera's configuration determined by the `temperature_sensor_setpoint`.
+        The `temperature_sensor_setpoint` defines a target or reference value for the camera's sensor, ensuring optimal or specific operation conditions for image acquisition.
+        Depending on the setpoint, the behavior or response of the camera sensor might vary.
+        """
+        return self.light.set(lf.AddIns.CameraSettings.SensorTemperatureSetPoint, deg_C)
+
+    def acquire_frame(self):
+        """
+        Acquires a frame (or series of frames) from the spectrometer, and the
+        corresponding wavelength data.
+        """
+        return self.light.acquire(), self.get_wavelengths()
+
+    def acquire_step_and_glue(self, wavelength_range):
+        """
+        Acquires a step and glue (wavelength sweep) over the specified range.
+        Wavelength range must have two elements (both in nm), corresponding
+        to the start and end wavelengths.
+        Please note that the wavelength must be calibrated for this to be useful.
+
+        Returns:
+        - tuple:
+        - spectrum (numpy.ndarray): The average spectrum obtained from the acquisition.
+        - wavelength (numpy.ndarray): An array of wavelengths corresponding to the spectrum.
+        """
+        lambda_min = wavelength_range[0]
+        lambda_max = wavelength_range[1]
+        try:
+            self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, True)
+        except Exception as e:
+            self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEnabled, False)
+            print(f'Unable to perform step and glue due to error: {e}')
+            return
+
+        self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueStartingWavelength, lambda_min)
+        self.light.set(lf.AddIns.ExperimentSettings.StepAndGlueEndingWavelength, lambda_max)
```

Comment:
Why don't we set these elsewhere like all of the other parameters?

Reply:
What do you mean by this?

Reply:
@cmordi  did we resolve this? My question was why do we set this to true, take data, then set it to false. Only to then have to set it to true again the next time we acquire data?